modules/auxiliary/gather/ldap_query.rb

```diff
@@ -0,0 +1,313 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+
+  include Msf::Exploit::Remote::LDAP
+  require 'json'
+  require 'yaml'
+
+  def initialize(info = {})
+    filename = 'ldap_queries_default.yaml'
+    user_config_file = File.join(::Msf::Config.get_config_root.to_s, filename)
+    unless File.exist?(user_config_file)
+      # If the user config file doesn't exist, then initialize it with the contents of the default one.
+      default_config_file = File.join(::Msf::Config.data_directory, 'auxiliary', 'gather', 'ldap_query', filename)
+      FileUtils.cp(default_config_file, user_config_file)
+    end
+
+    begin
+      @default_settings_file_path = user_config_file
+      @default_settings = YAML.safe_load(File.binread(@default_settings_file_path))
+    rescue StandardError => e
+      fail_with(Failure::BadConfig, "Couldn't parse #{@default_settings_file_path}, error was: #{e}")
+    end
+
+    unless @default_settings['queries']&.class == Array && !@default_settings['queries'].empty?
+      fail_with(Failure::BadConfig, "No queries supplied in #{@default_settings_file_path}!")
+    end
+
+    actions = []
+    @default_settings['queries'].each do |entry|
+      if entry['action'].nil? || entry['description'].nil?
+        if entry['action'].nil?
+          print_warning("Entry detected that was missing its 'action' and 'description' fields!")
+        else
+          print_warning("#{entry['Action']} is missing its 'description' field!")
+        end
+        print_warning("Please check the file at #{@default_settings_file_path} and fix the errors listed above!")
+        next
+      end
+      actions << [entry['action'], { 'Description' => entry['description'] }]
+    end
+    actions << ['RUN_QUERY_FILE', { 'Description' => 'Execute a custom set of LDAP queries from the JSON or YAML file specified by QUERY_FILE.' }]
+    actions << ['RUN_SINGLE_QUERY', { 'Description' => 'Execute a single LDAP query using the QUERY_FILTER and QUERY_ATTRIBUTES options.' }]
+    actions.sort!
+
+    default_action = 'RUN_QUERY_FILE'
+    if actions.length > 2 # Aka there is more than just RUN_QUERY_FILE and RUN_SINGLE_QUERY in the list...
+      default_action = actions[0][0] # Get the first entry's action name and set this as the default action.
+    end
+
```

Comment:
Could we put all of this into a method maybe? It looks like a code smell here 👀

Reply:
Sure can do, was looking for places to refactor the code into methods but didn't consider this location.

Reply:
Fixed with https://github.com/rapid7/metasploit-framework/pull/16598/commits/e6cdd04c0c1b08b55f13cc41527b1db2487f5a88 by moving this into its own method as requested.